src/Controller.cpp

```diff
@@ -84,7 +84,30 @@ void Controller::setFrameHandler(void (*fptr)(CANFrame *msg), byte opcodes[], by
 //
 void Controller::setParams(unsigned char *mparams)
 {
+  bool producer = 0;
+  bool consumer = 0;
+  bool nPreNormal = 0;
+  bool conOwn = 0;
+  byte mode = module_config->currentMode;
+  
   _mparams = mparams;
+  
+  for (Service * svc : services)
+  {
+    if (svc->getServiceID() == 5)
```

Comment:
Lots of anonymous constants and bit fiddling here.
What is serviceID 5 here? Use SERVICE_ID_PRODUCER.
See separate email on how to rewrite this function.